core/domain/html_validation_service.py

```diff
@@ -886,37 +867,48 @@ def add_dims_to_img_in_complex_rte(exploration_fs, html_string):
                 collapsible_component['content-with-value'] == ''):
             continue
         # Create a new soup with the content-with-value html string.
-        content_with_html_string = unescape_html(
-            collapsible_component['content-with-value'])
+        # Replace \n with '' because json.loads() cannot process it.
```

Comment:
But this seems wrong? Won't the end result have line breaks elided?

(I.e. please see if you can understand why the \n appears in the first place and whether removing the \n causes an error in reconstituting the original text.)

Reply:
I think you're right. Instead of removing it, I've found a way to escape it for json.loads() operation and then unescape it while saving it back.